MdePkg/Include/Register/Intel/StmApi.h

```diff
@@ -50,13 +58,29 @@ typedef struct {
   ///
   /// The total STM_HEADER should be 4K.
   ///
+
+  // Pad to take up 1KB of space
+  UINT8       Reserved2[SOFTWARE_STM_HEADER_PADDING_SIZE];
 } SOFTWARE_STM_HEADER;
 
 typedef struct {
-  MSEG_HEADER            HwStmHdr;
-  SOFTWARE_STM_HEADER    SwStmHdr;
+  UINT32    Signature;
+  UINT32    NumberOfCpus;
+  /// Pad to take up 1KB of space
+  UINT8     Reserved[CPU_INFORMATION_HEADER_PADDING_SIZE];
+} CPU_INFORMATION_HEADER;
+
+typedef struct {
+  MSEG_HEADER               HwStmHdr;
+  SOFTWARE_STM_HEADER       SwStmHdr;
+  CPU_INFORMATION_HEADER    CpuInfoHdr;
 } STM_HEADER;
 
+STATIC_ASSERT ((SIZE_1KB - OFFSET_OF (CPU_INFORMATION_HEADER, Reserved)) == CPU_INFORMATION_HEADER_PADDING_SIZE, "CPU_INFORMATION_HEADER_PADDING_SIZE mismatch!");

```

Comment:
nit: would you want to do `OFFSET_OF (STM_HEADER, SwStmHdr) = 1KB` instead? or just check the struct size of SOFTWARE_STM_HEADER to be 1KB?

Reply:
There are several ways to verify that we keep the headers at 1KB of size.  I'm fine with whatever we think is best.

Reply:
As long as the macro stays that holds the padding size, I'm also fine either way. Technically, we'd want to do both - `sizeof()` to check the total expected size including padding for each structure and `OFFSET_OF()` to check that the order of structs is maintained in `STM_HEADER`.

Reply:
I added the additional checks because I thought that's a better solution.